src/main/java/core/basesyntax/ArrayList.java

```diff
@@ -1,48 +1,97 @@
 package core.basesyntax;
 
+import java.util.Arrays;
+import java.util.NoSuchElementException;
+
 public class ArrayList<T> implements List<T> {
+    private static final int DEFAULT_CAPACITY = 10;
+    private Object[] array;
+    private int size;
+
+    public ArrayList() {
+        array = new Object[DEFAULT_CAPACITY];
+        size = 0;
+    }
+
     @Override
     public void add(T value) {
-
+        ensureCapacity(size + 1);
```

Comment:
not implemented task: array should grow 1.5 times when it is full

Reply:
but I increase the number of elements by a factor of one and a half in the ensureCapacity method`private void ensureCapacity(int minCapacity) {
        if (minCapacity > array.length) {
            int newCapacity = array.length + (array.length >> 1);
            if (newCapacity < minCapacity) {
                newCapacity = minCapacity;
            }
            T[] newArray = (T[]) new Object[newCapacity];
            System.arraycopy(array, 0, newArray, 0, size);
            array = newArray;
        }
    }`

Reply:
but I increase the number of elements by a factor of one and a half in the ensureCapacity method
<img width="801" alt="Screenshot 2024-02-20 at 16 47 02" src="https://github.com/mate-academy/jv-array-list/assets/146957474/3f1926c5-89f7-42a6-9e02-205b7ee10d1b">